dll/shellext/fontext/CFontCache.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * PROJECT:     ReactOS Font Shell Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     font list cache handling
+ * COPYRIGHT:   Copyright 2019 Mark Jansen (mark.jansen@reactos.org)
+ */
+
+#include "precomp.h"
+
+#define FONT_HIVE   HKEY_LOCAL_MACHINE
+#define FONT_KEY    L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Fonts"
```

Comment:
Store the value of `FONT_KEY` into a static variable.

Reply:
Why?

Reply:
Multiple instances will be generated by that. Make them a single instance.

Reply:
That is the compilers work, if the compiler cannot do that it shouldnt be used.

Reply:
I think technically the standard guarantees that different string literals have different addresses, so this optimization is not standards-compliant. That said, the compilers will usually do it anyway :p

Reply:
There are options for compilers in order to place strings in a string pool, so that same strings will end up being shared. This works OK as long as they are not shared betwen read-only data and possibly writable data...

Reply:
Even in debug builds there is one string in `.rdata` with this define:

![image](https://user-images.githubusercontent.com/610685/64924794-a01ab480-d7e8-11e9-9a9d-0e90526abc5f.png)

Reply:
That's correct. Indeed our cmake files explicitly specify using the string pool, hence you get only one string.